### PR TITLE
[Spark Doris Connector] Avoid consistency problem when has no more data

### PR DESCRIPTION
**Pull request by wuyunfeng**
When all result data returned, fetch_result blocked unexpected  by `blocking_get` after multi invoking .  Put sentinel nullptr for the BlockingQueue to keep consistent for repeated invoking fetch result.

### CODE REVIEW

**Review by EmmyMiao87 — APPROVED** [review]

**Review by EmmyMiao87 — DISMISSED** [review]

**Comment by EmmyMiao87** [comment]
Test for approve

File: be/src/runtime/result_queue_mgr.cpp [hunk 1]
```diff
@@ -47,6 +47,10 @@ Status ResultQueueMgr::fetch_result(const TUniqueId& fragment_instance_id, std::
         // sentinel nullptr indicates scan end
         if (*result == nullptr) {
             *eos = true;
+            // put sentinel for consistency, avoid repeated invoking fetch result when hava no rowbatch
+            if (queue) {
```
Comment (imay) [inline be/src/runtime/result_queue_mgr.cpp]:
```suggestion
            if (queue != nullptr) {
```

**Review by imay — APPROVED** [review]
LGTM

**Review by EmmyMiao87 — APPROVED** [review]
LGTM

**Comment by wuyunfeng** [comment]
https://github.com/apache/incubator-doris/issues/1525

**Pull request merged**